db/model/aspect.js

```diff
@@ -181,23 +171,27 @@ module.exports = function aspect(seq, dataTypes) {
        * TODO:
        * 1. Have a look at the sampleStore logic and confirm that it is
        * doing what it is supposed to do.
-       * 2. Wrap all the calls to redis in a promise and resolve it at the end
-       * like it has been done in the afterUpdate hook.
+       *
        */
 
       /**
        * When an aspect is created. Add its entry in the aspectStore
        * and the sampleStore if any.
        *
        * @param {Aspect} inst - The deleted instance
+       * @returns {Promise}
        */
       afterCreate(inst /* , opts */) {
+        const promiseArr = [];
+
         // Prevent any changes to original inst dataValues object
         const instDataObj = JSON.parse(JSON.stringify(inst.get()));
 
         // create an entry in aspectStore
-        redisOps.addKey(aspectType, inst.getDataValue('name'));
-        redisOps.hmSet(aspectType, inst.name, instDataObj);
+        promiseArr.push(redisOps.addKey(aspectType, inst.getDataValue('name')));
+        promiseArr.push(redisOps.hmSet(aspectType, inst.name, instDataObj));
+
+        return Promise.all(promiseArr);
```

Comment:
When the redis ops are right next to each other it would be cleaner to use a Promise.join. Same for afterDelete below.